test/e2e/init_scale_zero_test.go

```diff
@@ -67,56 +70,98 @@ func TestInitScaleZeroServiceLevel(t *testing.T) {
 func TestInitScaleZeroClusterLevel(t *testing.T) {
 	cancel := logstream.Start(t)
 	defer cancel()
 
 	clients := Setup(t)
-	namesCheckValidityUnset := test.ResourceNames{
-		Service: test.ObjectNameForTest(t),
-		Image:   "helloworld",
-	}
-	namesCheckValidityFalse := test.ResourceNames{
-		Service: test.ObjectNameForTest(t),
-		Image:   "helloworld",
-	}
-	namesCheckValidityTrue := test.ResourceNames{
-		Service: test.ObjectNameForTest(t),
-		Image:   "helloworld",
-	}
+	tests := []struct {
+		name                      string
+		// configDefaultsFlag indicated if the check validity on deploy flag is set to true
+		// or false in the config-defaults config map.
+		configDefaultsFlag        bool
+		// configAutoscalerFlag indicated if the check validity on deploy flag is set to true
+		// or false in the config-autoscaler config map.
+		configAutoscalerFlag      bool
+		// serviceCheckValidityUnset indicated whether pods will be created when the check
+		// validity on deploy flag is unset on the test service created.
+		serviceCheckValidityUnset bool
+		// serviceCheckValidityFalse indicated whether pods will be created when the check
+		// validity on deploy flag is set to false on the test service created.
+		serviceCheckValidityFalse bool
+		// serviceCheckValidityTrue indicated whether pods will be created when the check
+		// validity on deploy flag is set to true on the test service created.
+		serviceCheckValidityTrue  bool
+	}{{
+		name: "both config-defaults config-autoscaler false",
+		configDefaultsFlag: false,
+		configAutoscalerFlag: false,
+		serviceCheckValidityUnset: false,
+		serviceCheckValidityFalse: false,
+		serviceCheckValidityTrue: true,
+	}, {
+		name: "config-autoscaler true but config-defaults false",
+		configDefaultsFlag:        false,
+		configAutoscalerFlag:      true,
+		serviceCheckValidityUnset: true,
+		serviceCheckValidityFalse: true,
+		serviceCheckValidityTrue:  true,
+	}, {
+		name: "config-defaults true but config-autoscaler false",
+		configDefaultsFlag: true,
+		configAutoscalerFlag: false,
+		serviceCheckValidityUnset: true,
+		serviceCheckValidityFalse: false,
+		serviceCheckValidityTrue: true,
+	}, {
+		name: "both config-defaults config-autoscaler true",
+		configDefaultsFlag: true,
+		configAutoscalerFlag: true,
+		serviceCheckValidityUnset: true,
+		serviceCheckValidityFalse: true,
+		serviceCheckValidityTrue: true,
+	}}
+	for _, tc := range tests {
+		func() {
+			t.Logf("Setting feature flag to %v on config-defaults ConfigMap and %v on config-autoscaler ConfigMap.", tc.configDefaultsFlag, tc.configAutoscalerFlag)
+			updatedConfigDefaultsCM := setCheckValidityOnDeployOnCluster(t, clients, "config-defaults", tc.configDefaultsFlag)
+			updatedConfigAutoscalerCM := setCheckValidityOnDeployOnCluster(t, clients, "config-autoscaler", tc.configAutoscalerFlag)
+			updatedConfigDefaultsCM.Data["check-validity-on-deploy"] = "true"
+			updatedConfigAutoscalerCM.Data["check-validity-on-deploy"] = "true"
```

Comment:
should be able extract as a func call

Reply:
actually it's not necessary to set it back to true because `setCheckValidityOnDeployOnCluster(...)` already does it at the end